lib/puppet/context/trusted_information.rb

```diff
@@ -51,17 +61,42 @@ def self.remote(authenticated, node_name, certificate)
           [ext['oid'].freeze, ext['value'].freeze]
         end]
       end
-      new('remote', node_name, extensions)
+      new('remote', node_name, extensions, external)
     else
-      new(false, nil, {})
+      new(false, nil, {}, external)
```

Comment:
If `trusted_external_command` is not defined, then `retrieve_trusted_external` will return nil, which is passed in here and above on line 64. As a result `Puppet::Context::TrustedInformation#external` may return nil or a Hash. I'm thinking it would be better if `retreive_trusted_external` always returned a `Hash`